cloudinary.go

```diff
@@ -0,0 +1,390 @@
+	return true
+}
+
+// Files returns all the files from the given directory.
+func (r *Cloudinary) Files(path string) ([]string, error) {
+	folders, err := r.instance.Admin.Search(r.ctx, search.Query{
+		Expression: fmt.Sprintf("folder:%s", validPath(path)),
+		SortBy: []search.SortByField{
+			{"public_id": search.Ascending},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	var result []string
+	for _, folder := range folders.Assets {
+		result = append(result, folder.PublicID)
+	}
+	return result, nil
+}
+
+// Get returns the contents of a file.
+func (r *Cloudinary) Get(file string) (string, error) {
+	rawContent, err := GetRawContent(r.Url(file))
+	if err != nil {
+		return "", err
+	}
+	return string(rawContent), nil
+}
+
+// LastModified returns the last modified time of a file.
+func (r *Cloudinary) LastModified(file string) (time.Time, error) {
+	resource, err := r.getAsset(file)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return resource.CreatedAt, nil
+}
+
+// MakeDirectory creates a directory.
+func (r *Cloudinary) MakeDirectory(directory string) error {
+	result, err := r.instance.Admin.CreateFolder(r.ctx, admin.CreateFolderParams{
+		Folder: directory,
+	})
+	if err != nil {
+		return err
+	}
+	if !result.Success {
+		return fmt.Errorf("make directory error: %+v", result.Error)
+	}
+	return nil
+}
+
+// MimeType returns the mime-type of a file.
+func (r *Cloudinary) MimeType(file string) (string, error) {
+	resource, err := r.getAsset(file)
+	if err != nil {
+		return "", err
+	}
+	// Check if the resource format is empty, return only the resource type.
+	if resource.Format == "" {
+		return resource.ResourceType, nil
+	}
+	// Replace 'jpg' with 'jpeg' in the format if it is 'jpg'
+	format := strings.ReplaceAll(resource.Format, "jpg", "jpeg")
+
+	return resource.ResourceType + "/" + format, nil
+}
+
+// Missing checks if a file is missing.
+func (r *Cloudinary) Missing(file string) bool {
+	return !r.Exists(file)
+}
+
+// Move moves a file to a new location.
+func (r *Cloudinary) Move(source, destination string) error {
+	rename, err := r.instance.Upload.Rename(r.ctx, uploader.RenameParams{
+		FromPublicID: r.getPublicId(source),
+		ToPublicID:   r.getPublicId(destination),
+		ResourceType: r.getResourceType(destination),
+	})
+	if err != nil {
+		return err
+	}
+	if rename.Error != nil {
+		return fmt.Errorf("move file error: %#v", rename.Error)
+	}
+	return nil
+}
+
+// Path returns the full path for a file.
+func (r *Cloudinary) Path(file string) string {
+	return validPath(file)
+}
+
+// Put stores a new file on the disk.
+func (r *Cloudinary) Put(file, content string) error {
+	tempFile, err := r.tempFile(content)
+	defer os.Remove(tempFile.Name())
+	if err != nil {
+		return err
+	}
+	_, err = r.instance.Upload.Upload(r.ctx, tempFile.Name(), uploader.UploadParams{
+		PublicID:       r.getPublicId(file),
+		UseFilename:    api.Bool(true),
+		UniqueFilename: api.Bool(false),
+		ResourceType:   r.getResourceType(file),
```

Comment:
Maybe `ResourceType` is not required, see the official example: https://github.com/cloudinary/cloudinary-go/blob/bb1164bac6075db30220761632d2b4e456c7f629/cloudinary_test.go#L47C1-L47C1

Please check that other `ResourceType` is also not required.